lib/requester/core.js

```diff
@@ -275,11 +277,25 @@ module.exports = {
             requestMethod = (typeof request.method === 'string') ? request.method.toLowerCase() : undefined,
             bodyIsEmpty = requestBody.isEmpty(),
             bodyIsDisabled = requestBody.disabled,
-            bodyContent = requestBody[requestBodyType];
+            bodyContent = requestBody[requestBodyType],
+            protocolProfileBehavior = requestBody.protocolProfileBehavior,
+            disableBodyPruning = protocolProfileBehavior === 'disable-body-pruning';
 
         // early bailout for empty or disabled body (this area has some legacy shenanigans)
         // @todo support a body type where nothing is sent and remove this GET restriction
-        if (bodyIsEmpty || bodyIsDisabled || (requestMethod === GET && !options.sendBodyWithGetRequests)) {
+        if (bodyIsEmpty || bodyIsDisabled) {
+            return;
+        }
+
+        // bail out if request method doesn't support body and disableBodyPruning is not set.
+        if (_.includes(METHODS_WITHOUT_BODY, requestMethod) && protocolProfileBehavior && !disableBodyPruning) {
```

Comment:
Also, it's always being used in a negated form, so adjust your initialization accordingly.

Reply:
Use hash.